### PR TITLE
AirbyteLib: Require stream selection

### PR DESCRIPTION
**Pull request by aaronsteers**
With this change, users will be directed to call "select_streams()" or "select_all_streams()" if they attempt to call "source.read()" without having already done so.

Resolves:
- https://github.com/airbytehq/airbyte-lib-private-beta/issues/17

This also renames "set_streams()" to "select_streams()" to align with the 'select' verb and 'selected' decriptor used elsewhere.

I've kept 'set_streams()' alive and still works, but it will now emit a `DeprecationWarning` when called, instructing users they should have called  `select_streams()`.

I've also brought back `get_connector()` as an alias of `get_source()` using the same deprecation warning approach.

### CODE REVIEW

File: airbyte-lib/airbyte_lib/source.py [hunk 1]
```diff
@@ -252,19 +264,24 @@ def configured_catalog(self) -> ConfiguredAirbyteCatalog:
...
             streams=[
-                # TODO: Set sync modes and primary key to a sensible adaptive default
                 ConfiguredAirbyteStream(
                     stream=stream,
-                    sync_mode=SyncMode.incremental,
                     destination_sync_mode=DestinationSyncMode.overwrite,
                     primary_key=stream.source_defined_primary_key,
+                    # TODO: The below assumes all sources can coalesce from incremental sync to
+                    # full_table as needed. CDK supports this, so it might be safe:
+                    sync_mode=SyncMode.incremental,
```
Comment (aaronsteers) [inline airbyte-lib/airbyte_lib/source.py:281]: This is a drive-by comment-revamp, not directly related to this PR.

**Review by bindipankhudi — APPROVED** [review]
looks good to me!
